### PR TITLE
Fixes#1425: Add Tests for chat_message.dart

### LINKED ISSUE #1425: Models: Tests for chat_message.dart
- Please coordinate **issue assignment** and **PR reviews** with the contributors listed in this issue https://github.com/PalisadoesFoundation/talawa/issues/359

We will need unittests done for all methods, classes and functions found in this file.

Any widgets referenced in this file must also have unittests done

```
lib/models/chats/chat_message.dart
```

- Parent Issue: https://github.com/PalisadoesFoundation/talawa/issues/1414

### PR Acceptance Criteria

- When complete this file must show **100%** coverage when merged into the code base.
- [The current code coverage for the file can be found here](https://codecov.io/gh/PalisadoesFoundation/talawa/tree/develop/lib/models/chats)
- The PR will show a report for the code coverage for the file you have added. You can use that as a guide

### PR DESCRIPTION
**Pull request by antonio-pedro99**
**What kind of change does this PR introduce?**

Made the ChatMessage to be nested serializable by adding@JsonSerializable(explicitToJson: true).
Add tests for ChatMessage

**Issue Number:**

Fixes #1425 

**Did you add tests for your changes?**

Yes!

**Summary**

Add Tests for ChatMessage model.

**Does this PR introduce a breaking change?**

No

**Other information**

I had to change the pubspec.yaml due to problems related to build_runner, I added analyzer and json_serializable under dev_dependencies.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Review by antonio-pedro99 — COMMENTED** [review]
we need to pass explicitToJson to true in order to have nested class serialization.

**Comment by antonio-pedro99** [comment]
@noman2002 can you review this PR?

File: lib/models/chats/chat_message.g.dart [hunk 1]
```diff
@@ -6,23 +6,21 @@ part of 'chat_message.dart';
...
 
 Map<String, dynamic> _$ChatMessageToJson(ChatMessage instance) =>
     <String, dynamic>{
-      '_id': instance.id,
-      'sender': instance.sender,
-      'receiver': instance.receiver,
+      'id': instance.id,
+      'sender': instance.sender?.toJson(),
+      'receiver': instance.receiver?.toJson(),
       'messageContent': instance.messageContent,
```
[the remark this answers is not in the document]
Comment (antonio-pedro99) [reply lib/models/chats/chat_message.g.dart:25]: as I said, in order to have nested serialization, we need to make explicitToJson true in @JsonSerializable() annotation. the reason why it changed to from _id to id I do not really know, but the reason we have .toJson() in line 23 and 24 is that, when I will can ChatMessage's toJson() method, all toJson() method of any class nested(in this case ChatUser) in _$ChatMessageToJson method will be invoked.

[the remark this answers is not in the document]
Comment (antonio-pedro99) [reply lib/models/chats/chat_message.g.dart:25]: in really it is alread "id" and not "_id". 
- represents remotion and + addition!
from line 22 to 24  and from line 9 to 18 is what I have changed.

[the remark this answers is not in the document]
Comment (noman2002) [reply lib/models/chats/chat_message.g.dart:25]: ops sorry. I mean change it back to **_id**

[the remark this answers is not in the document]
Comment (antonio-pedro99) [reply lib/models/chats/chat_message.g.dart]: It was not intentional. by the way, this message is auto generated. I can not understand why it changed if I have only added @JsonSerializable(explicitToJson:true) in ChatMessage class.

**Review by noman2002 — APPROVED** [review]

**Review by palisadoes — APPROVED** [review]

**Pull request merged**